plugin/pkg/shoot/managedseed/admission.go

```diff
@@ -160,6 +165,10 @@ func (v *ManagedSeed) validateUpdate(ctx context.Context, a admission.Attributes
 		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "kubernetes", "verticalPodAutoscaler", "enabled"), vpaEnabled, "shoot VPA has to be enabled for managed seeds"))
 	}
 
+	if oldShoot.Spec.Networking.Nodes != nil && *oldShoot.Spec.Networking.Nodes != *shoot.Spec.Networking.Nodes {
```

Comment:
Same question like above, I guess. Why are `ManagedSeed`s different?

Reply:
This is a new check for managed seeds. We don't want that the nodes range changes, as it might introduce an overlap to a shoot.
Without the feature gate,  this additional check should not make a difference.

Reply:
Instead of "generally" forbidding it, shouldn't we rather list all shoots on this (managed) seed and check whether there would be an overlap, and only deny in such case? If the expansion is valid, we could allow it.

Reply:
Ideally that would be the case. When we discussed the implementation at first we wanted to avoid race conditions during scheduling of new shoots or potentially concurrent updates so we deferred verifying that for later. But if you see no danger we can reconsider.

Reply:
Such race condition is technically possible, of course, but it's also VERY unlikely, don't you think? 😉 I would definitely trade this in for the feature of changing the CIDR also for managed seeds. :)

Reply:
Listing all dependent shoots for an overlap validation can be done. But here we are validating the `shoot` resource. If we change this field in the shoot, the `managedseed` resource seems not to be updated. If we update the `managedseed` resource manually, it will check the `seed` resource. But in the resource `seed` the field is immutable. Making it mutable in the seed, we would need to add overlap checks to all shoot networks to the `seed` update validation.
I didn't look into how `managedseed`s could be automatically updated upon a change of the registered shoot. 
Overall, I would guess that the effort for implementing all this, is a bit of a feature creep considering, that we don't need the field to be mutable in `shoot`s that are registered in `managedseed`s.
But may be I miss something, here.

Reply:
Generally, I think we would need to adapt the [`gardenlet`'s `ManagedSeed` controller](https://github.com/gardener/gardener/blob/master/pkg/gardenlet/controller/managedseed/add.go) to also react on relevant changes in the `Shoot`s to make this work.

> considering, that we don't need the field to be mutable in shoots that are registered in managedseeds.

What do you mean by this? Why don't we need this functionality?

Reply:
The current requests for a mutable `networks.Nodes` field comes from customers, that need it for shoots that are not seeds. I don't say that it wouldn't be useful for shoots that are seeds. But, I don't see the current need for it.

Reply:
Fair enough. Given that the change is not too straight-forward, we can also keep it for a separate PR and proceed with this single change for now.